pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1698,6 +1856,39 @@ impl Timeline {
             let img = tline.get(key, tline.ancestor_lsn, ctx).await?;
             Ok(Some((key, tline.ancestor_lsn, img)))
         }
+        let image_layer_key = PersistentLayerKey {
+            key_range: hack_image_layer_range,
+            lsn_range: PersistentLayerDesc::image_layer_lsn_range(lowest_retain_lsn),
+            is_delta: false,
+        };
+
+        let discard_image_layer = {
```

Comment:
```suggestion
        // Like with delta layers, it can happen that we re-produce an already existing image layer.
        // TODO: explain why it's safe to skip.
        let discard_image_layer = {
```

Reply:
Explained. Currently it only happens with manual force compaction.